[FEATURE] #102793 - PSR-14 event for modifying default constraints in PageRepository (#3878)
TYPO3-Documentation/TYPO3CMS-Reference-CoreApi: Documentation/ApiOverview/Events/Events/Core/Domain/ModifyDefaultConstraintsForDatabaseQueryEvent.rst

..  include:: /Includes.rst.txt
..  index:: Events; ModifyDefaultConstraintsForDatabaseQueryEvent
..  _ModifyDefaultConstraintsForDatabaseQueryEvent:

=============================================
ModifyDefaultConstraintsForDatabaseQueryEvent
=============================================

..  versionadded:: 13.0
    This event serves as a replacement for the removed hook
    :php:`$GLOBALS['TYPO3_CONF_VARS']['SC_OPTIONS']['t3lib/class.t3lib_page.php']['addEnableColumns']`.

> The API class :php:`\TYPO3\CMS\Core\Domain\Repository\PageRepository` has a
> method :php:`getDefaultConstraints()` which accumulates common restrictions for
> a database query to limit a query for TCA-based tables in order to filter out
> disabled or scheduled records.

Reviewer: ```suggestion
The API class :php:`\TYPO3\CMS\Core\Domain\Repository\PageRepository` has a
method :php:`getDefaultConstraints()` which accumulates common 
restrictions for a database query. The purpose is to limit queries for 
TCA-based tables, filtering out disabled or scheduled records.
```
I would avoid very long sentences
